docs/how-to/mgc-setup.md

```diff
@@ -0,0 +1,145 @@
+- [Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- [Kustomize](https://kubectl.docs.kubernetes.io/installation/kustomize/)
+- [Helm](https://helm.sh/docs/intro/quickstart/#install-helm)
+
+## Configure OCM with RawFeedbackJsonString Feature Gate
+
+All OCM spoke clusters must be configured with the `RawFeedbackJsonString` feature gate set to true. This can be done in two ways:
+
+1. When running the `clusteradm join` command that joins the spoke cluster to the hub:
+
+```bash
+clusteradm join <snip> --feature-gates=RawFeedbackJsonString=true
+```
+
+2. By patching each spoke cluster's `klusterlet` in an existing OCM install:
+
+```bash
+kubectl patch klusterlet klusterlet --type merge --patch '{"spec": {"workConfiguration": {"featureGates": [{"feature": "RawFeedbackJsonString", "mode": "Enable"}]}}}' --context <EACH_SPOKE_CLUSTER>
+```
+
+## Installing MGC
+
+Run the following command in the context of your hub cluster to install the MGC [TODO: change build to Kuadrant/MGC repo/main]:
+
+```bash
+kustomize --load-restrictor LoadRestrictionsNone build "github.com/eoinfennessy/multicluster-gateway-controller.git/config/mgc-install-guide?ref=mgc_setup_guide_for_ocm" --enable-helm | kubectl apply -f -
+```
+
+In addition to the MGC, this will also install the Kuadrant add-on manager, Gateway API CRDs, and [cert-manager](https://cert-manager.io/).
+
+After the configuration has been applied, you can verify that the MGC and add-on manager have been installed and are running:
+
+```bash
+kubectl get pods -n multicluster-gateway-controller-system
+```
+```
+NAME                                           READY   STATUS    RESTARTS   AGE
+mgc-controller-manager-8659cfdf46-nwn4w        1/1     Running   0          2m
+mgc-kuadrant-add-on-manager-7479cb7f9c-rzk4w   1/1     Running   0          2m
+```
+
+## Creating a ManagedZone
+
+To manage the creation of DNS records, MGC uses [ManagedZone](https://docs.kuadrant.io/multicluster-gateway-controller/docs/how-to/managedZone/) resources. A `ManagedZone` can be configured to use DNS Zones on either AWS (Route53), and GCP. We will now create a ManagedZone on the cluster using AWS credentials.
+
+First, export the [environment variables detailed here](https://docs.kuadrant.io/multicluster-gateway-controller/docs/getting-started/#config) in a terminal session.
+
+Next, create a secret and configMap containing the AWS credentials and config [TODO: is the ConfigMap needed here?]:
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mgc-aws-credentials
+  namespace: multicluster-gateway-controller-system
+type: "kuadrant.io/aws"
+stringData:
+  AWS_ACCESS_KEY_ID: ${MGC_AWS_ACCESS_KEY_ID}
+  AWS_SECRET_ACCESS_KEY: ${MGC_AWS_SECRET_ACCESS_KEY}
+  AWS_REGION: ${MGC_AWS_REGION}
+EOF
+
+cat <<EOF | kubectl apply -f -
```

Comment:
hmm are these actually needed?

Reply:
I don't think this configmap is needed

Reply:
The secret is used for the `dnsProviderSecretRef` in the `ManagedZone` config. I'm not sure, but I think we may be able to do without the configMap as the zone root domain & ID is specified in the `ManagedZone` spec.

Reply:
Yeah I tested without and it worked